mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -1401,6 +1413,17 @@ impl<'a> TakerOrderBuilder<'a> {
         self
     }
 
+    /// When a new [TakerOrderBuilder::new] is created, it sets [SWAP_VERSION] by default.
+    /// However, if user has not specified in the config to use TPU V2,
+    /// the TakerOrderBuilder's swap_version is changed to legacy.
+    /// In the future alls users will be using TPU V2 by default without "use_trading_proto_v2" configuration.
+    pub fn use_trading_proto_v2(mut self, use_trading_proto_v2: bool) -> Self {
+        if !use_trading_proto_v2 {
+            self.swap_version = legacy_swap_version();
+        }
+        self
+    }
```

Comment:
Why not this?

```suggestion
    pub fn use_trading_proto_v2(mut self) -> Self {
        self.swap_version = legacy_swap_version();
        self
    }
```

or, at least rename it to something like `maybe_**`

Reply:
u suggested to change version to legacy protocol calling `use_trading_proto_v2`
please re read doc comment
```rust
    /// When a new [TakerOrderBuilder::new] is created, it sets [SWAP_VERSION] by default.
    /// However, if user has not specified in the config to use TPU V2,
    /// the TakerOrderBuilder's swap_version is changed to legacy.
    /// In the future alls users will be using TPU V2 by default without "use_trading_proto_v2" configuration.
    pub fn use_trading_proto_v2(mut self, use_trading_proto_v2: bool) -> Self {
```

Reply:
What I suggest is, either remove this boolean from the function and check it on the caller side or declare that in the function name with "maybe" prefix.

Reply:
> What I suggest is, either remove this boolean from the function and check it on the caller side

I dont get it, on the caller side it is where? Right now it looks like, you call `use_trading_proto_v2` and set legacy
```
    pub fn use_trading_proto_v2(mut self) -> Self {
        self.swap_version = legacy_swap_version();
        self
    }
```

Reply:
I updated function name https://github.com/KomodoPlatform/komodo-defi-framework/pull/2324/commits/e37c74e9dcd27a3022734e3e1ff25fa8537fa69e

Reply:
I didn't pay attention to the logic, the usage was just weird. Looking at the logic, it's even more weird.. Function is called `use_trading_proto_v2` but it downgrades version to legacy if given boolean is false.

Why not just this:
```rs
    pub fn use_trading_proto_legacy(mut self) -> Self {
        self.swap_version = legacy_swap_version();
        self
    }
```

and call it only when `use_trading_proto_v2` is false?

Reply:
the thing is that use_trading_proto_v2 initially is a config boolean flag, so use_trading_proto_legacy is supposed to be a question

ok I refactored function, now its purpose is only to set legacy protocol https://github.com/KomodoPlatform/komodo-defi-framework/pull/2324/commits/a62f56aebfe57329e1c880ff2f58ee3f620697bb